### PR TITLE
Amine Bensalem

### PR DESCRIPTION
**Pull request by AminesCodes**
Answered String Lab questions

### CODE REVIEW

File: String_Lab.js [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+// 4. Create a code block that takes a single string variable, and returns
+// a copy of the string with the last letter capitalized. For example:
+
+// 'jimmy'
+// => 'jimmY'
+console.log(`Capitalizing the last character of the word ${string} :`);
+console.log("Methode1 : ")
+console.log(string.slice(0, string.length - 1) + string[string.length - 1].toUpperCase());
+console.log("Methode2 : ")
+console.log(`${string}\b${string[string.length - 1].toUpperCase()}`);
```
Comment (alejo4373) [inline String_Lab.js]: The use of `\b` (backspace) for string manipulation yields the result we are looking for and I'm glad you found such special character but is not used conventionally. Think of the result of this expression `"Lesson\b\b".length`

File: String_Lab.js [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+// the ' dollars', your code block should add '(pinky)' at the end of
+// the amount if it's 1 million. For example:
+
+// let amount = 10
+// => 10 dollars
+// let amount = 1000000
+// =>  1000000 dollars (pinky)
+let amount = 1000000;
+if (amount === 1000000) {
+console.log(`Amount of money :\n\t ${amount} dollars (pinky) ... Wish I had that in my bank account :)`);
```
Comment (alejo4373) [inline String_Lab.js]: Lines inside a code block (body) of an `if`, `for` or `while` statements should be indented to the right to indicate that they are inside something else (the body/code block of a statement)

File: String_Lab.js [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+} else if (amount === 1 || amount === 0) {
+console.log(`Amount of money :\n\t ${amount} dollar`);
+} else console.log(`Amount of money :\n\t ${amount} dollars`);
```
Comment (alejo4373) [inline String_Lab.js]: This `else` and other statements whose body only contains another single statement doesn't require `{}` (curly braces) to indicate a body or code block. It is still my personal preference to include them.

File: String_Lab.js [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+// verbing('swimming')
+// =>  'swimmingly'
+// verbing('go')
+// =>  'go'
+let verb = "train";
+console.log(`Verbing the word ${verb} : `)
+if (verb.length >=3) {
+   if (verb.slice(verb.length - 3) === "ing") {
+      console.log(verb + "ly");
+   } else console.log(verb + "ing");
```
Comment (alejo4373) [inline String_Lab.js]: Same as above single statement code block doesn't require `{}`  but I would still include them

File: String_Lab.js [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+for (let i = 0; i < string.length - 1; i++){
+  withoutLast += string[i];
+}
+console.log(`Methode2 : ${withoutLast}`);
+console.log(`Methode3 : ${string}\b `);
+
+//let last = string[string.length - 1];
+//console.log("Methode4 : ", string.split(last));
+
+let str3 = "Had some hard time with this part"
```
Comment (alejo4373) [inline String_Lab.js]: I'm glad you got it. 👍

File: String_Lab.js [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+} else console.log(`Error : ${sentence} is too short for this operation`);
+console.log("---------------------------------- \n");
+
+
+//Bonus3:
+//Write a code block that prints/logs whether a string x starts with a vowel.
+
+let x = "Engine";
+
+switch (x[0].toLowerCase()){
```
Comment (alejo4373) [inline String_Lab.js]: Nice work!

File: String_Lab.js [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+
+//Bonus4:
+//Write a code block that given two strings, str1 and str2,
+//prints/logs a new string with str2 mashed into the middle of str1
+//Examples:
+//For str1 = 'help' and str2 = 'me'. Your program should log 'hemelp'
+//For str1 = 'hello' and str2 = 'world'. Your program should log 'helworldlo'
+
+let str1 = "hello", str2 = "world"; //Your program should log 'hemelp'
+let part1 = str1.slice(0, Math.ceil(str1.length / 2));
```
Comment (alejo4373) [inline String_Lab.js]: Nice, now you know `Math.ceil` and `Math.floor`. Clever application here specially of `Math.ceil`

Comment (AminesCodes) [reply String_Lab.js]: Thanks a lot, appreciate all your feedback

**Pull request closed without merging**